src/store/modules/management.js

```diff
@@ -108,6 +117,13 @@ const mutations = {
     state.managedFunctions = functionsUpdatedNotification.functions;
     if (state.selectedFunction === null) {
       state.selectedFunction = state.managedFunctions[0];
+    } else {
+      let functionsWithId = state.managedFunctions.filter(s => s.id === state.selectedFunction.id);
+      if (functionsWithId.length > 0) {
+        state.selectedFunction = functionsWithId[0];
+      } else {
+        state.selectedFunction = state.managedFunctions[0];
+      }
     }
   },
   [M.SELECT_MANAGED_TRUCK]({ commit }, managedTruck) {
```

Comment:
the mutator parameter "managedTruck" is never used. it should be removed.

Reply:
but it is used :(

Reply:
Sorry, my appologies, I should have read the entire file here